### DISCUSSION TITLE
Retrieving specific fields from the filtered results is not ordered

### DISCUSSION THREAD

**Discussion by jmdivya** [ohler55/ojg#109] [Q&A] [answered]
Hi, The Library is pretty good.Thanks.
I faced an issue and looking for your suggestion. If I use the below expression on the sample json given
Expr1 : $.Books[?(@.type=='Humor')]  ==> This expression retrieves the result in order i.e details of book1 followed by book2. So If I define some methods like getFirst, getLast -I am able to retrieve the correct element.
But for Expr : $.Books[?(@.type=='Humor)].name ==> Using name after the filter causes the o/p in reverse order. So I get the o/p as Book2, Book1. Am I missing something, is it possible to achieve the order when I filter the filtered result with json fields?

{
"Books" :[
{
"name": "Book1"
"type": "Humor"
},
{
"name": "Book2"
"type": "Action"
},
{
"name": "Book3"
"type": "Humor"
}

]
}

**Comment by ohler55** [comment]
Order should be the same. It's a bug. I'll get it fixed.

**Comment by ohler55** [comment]
Please see if the develop branch fixes the issue.

**Reply by jmdivya** [reply]
It fixes the issue. Thanks for the quick fix

**Reply by ohler55** [reply, marked as answer]
Better to address issues early if possible. v1.17.3 released with the fix. Thanks for pointing out the issue.
